cartridge/roles.lua

```diff
@@ -242,10 +243,9 @@ end
 -- @treturn[2] nil
 -- @treturn[2] table Error description
 local function validate_config(conf_new, conf_old)
-    if type(conf_new) ~= 'table'  then
-        return nil, ValidateConfigError:new('config must be a table')
-    end
     checks('table', 'table')
+    assert(conf_new.__type ~= 'ClusterwideConfig')
```

Comment:
I'm puzzled a bit why you check that this isn't a `ClusterwideConfig`. At least write a comment what you are defending against.